### PR TITLE
Introduce stubdom recipes

### PR DESCRIPTION
**Pull request by dpsmith**
This PR enables packaging of the of stub domains so they can be incorporated under IMAGE_INSTALL and then converts dom0 image recipe.

### CODE REVIEW

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6737

File: recipes-openxt/stubdomains/devicemodel-stubdom_1.0.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+FILESEXTRAPATHS_prepend := "${DEPLOY_DIR}/images/${STUBDOMAIN_MACHINE}:"
+
+SRC_URI = " \
+        file://xenclient-stubdomain-initramfs-image-${STUBDOMAIN_MACHINE}.cpio.gz;unpack=0 \
+"
+
+do_install() {
+        install -d ${D}/usr/lib/xen/boot
+        install -m 0644 ${WORKDIR}/xenclient-stubdomain-initramfs-image-${STUBDOMAIN_MACHINE}.cpio.gz \
+            ${D}/usr/lib/xen/boot/stubdomain-initramfs
```
[the remark this answers is not in the document]
Comment (dpsmith) [reply recipes-openxt/stubdomains/devicemodel-stubdom_1.0.bb]: ack

**Comment by dpsmith** [comment]
> Since there is no dependency system between `MACHINE` targets in bitbake, would it be difficult to add a `bbfatal` message in `do_fetch` if the stub-domain kernel or initramfs image are not built yet? It would make the error more explicit.

After discussion I think we have arrived at a good solution, will push shortly.

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6749

**Comment by jean-edouard** [comment]
Merging soon.

**Pull request merged**